### ISSUE TITLE
FormattedDate Type is missing the children prop with React 18 types

### ISSUE THREAD

**Issue by scottdickerson** [formatjs/formatjs#3851]
## Which package?
```
 "react-intl": "^6.1.1",
"@types/react": "18.0.20"
```

## Describe the bug
Typescript errors in FormattedDate and FormattedTime missing the children property:
Here's the error from `tsc`
```
error TS2322: Type '{ children: (createdText: any) => Element; month: "short"; day: "numeric"; year: "numeric"; value: Date; }' is not assignable to type 'IntrinsicAttributes & DateTimeFormatOptions & CustomFormatConfig<string> & { value: string | number | Date | undefined; }'.
  Property 'children' does not exist on type 'IntrinsicAttributes & DateTimeFormatOptions & CustomFormatConfig<string> & { value: string | number | Date | undefined; }'.
```

See this type definition
```
export declare const FormattedDate: React.FC<Intl.DateTimeFormatOptions & CustomFormatConfig & {
    value: string | number | Date | undefined;
}>;
```

React FC removed the children prop in React 18 types

Notice the documentation says the children prop is explicitly specified in the type:
https://formatjs.io/docs/react-intl/components/#formatteddate
<img width="986" alt="image" src="https://user-images.githubusercontent.com/6663002/191539971-782a3e4c-9d7e-4c69-bd52-30e37151fc6e.png">

### Codesandbox URL

I don't see this in codesandbox, but I do have a repo

### Reproducible Steps/Repo

https://github.com/scottdickerson/react-intl-test

Open this in VSCode and you should see the type error.
Or run `npm start`

## Expected behavior
Passing correct typescript type information as it's preventing our builds

## Desktop (please complete the following information):

- OS: OS X
- Browser chrome
- Version 6.1.1

**Comment by scottdickerson** [comment]
possible fix here: https://github.com/formatjs/formatjs/pull/3852.  It works for me locally.

To verify use @types/react: 18.0.20

**Comment by scottdickerson** [comment]
this is fixed now in in the 6.1.2 release of `react-intl`
